backend/app/api/v1/endpoints/cpt/cptJobs.py

```diff
@@ -0,0 +1,134 @@
+            "clusterName": "ci-op-4n0msnvp-7904a-s5sv8",
+            "ocpVersion": "4.14.0-0.nightly-2023-09-15-233408",
+            "networkType": "OVNKubernetes",
+            "buildTag": "1704299395064795136",
+            "jobStatus": "success",
+            "buildUrl": "https://example.com/1704299395064795136",
+            "upstreamJob": "periodic-ci-openshift",
+            "upstreamJobBuild": "5fe07ad3-5415-433c-b9af-f60545d0d432",
+            "executionDate": "2023-09-20T02:14:07Z",
+            "jobDuration": "5261",
+            "startDate": "2023-09-20T02:14:07Z",
+            "endDate": "2023-09-20T03:41:48Z",
+            "timestamp": "2023-09-20T02:14:07Z",
+            "shortVersion": "4.14"
+        },
+        {
+            "ciSystem": "PROW",
+            "uuid": "CPT-0d58dddf-721a-4952-985e-046bc17ee3cc",
+            "releaseStream": "4.13.0-0.nightly",
+            "platform": "GCP",
+            "clusterType": "self-managed",
+            "benchmark": "node-density",
+            "masterNodesCount": 3,
+            "workerNodesCount": 24,
+            "infraNodesCount": 3,
+            "masterNodesType": "e2-standard-4",
+            "workerNodesType": "e2-standard-4",
+            "infraNodesType": "n1-standard-16",
+            "totalNodesCount": 30,
+            "clusterName": "ci-op-x2ic4nsf-8360f-kzbcg",
+            "ocpVersion": "4.13.0-0.nightly-2023-09-12-074803",
+            "networkType": "OVNKubernetes",
+            "buildTag": "1704367060252889088",
+            "jobStatus": "success",
+            "buildUrl": "https://example/1704367060252889088",
+            "upstreamJob": "periodic-ci-openshift",
+            "upstreamJobBuild": "3ab02e3b-3000-4fc9-a30c-9cd02fe4a78c",
+            "executionDate": "2023-09-20T07:19:00Z",
+            "jobDuration": "582",
+            "startDate": "2023-09-20T07:19:00Z",
+            "endDate": "2023-09-20T07:28:42Z",
+            "timestamp": "2023-09-20T07:19:00Z",
+            "shortVersion": "4.13"
+        },
+    ]
+}
+
+unprocessable_example = {"error": "invalid date format, start_date must be less than end_date"}
+
+@router.get('/api/cpt/v1/jobs',
+            summary="Returns a job list from all the products.",
+            description="Returns a list of jobs in the specified dates. \
+            If not dates are provided the API will default the values. \
+            `startDate`: will be set to the day of the request minus 5 days.\
+            `endDate`: will be set to the day of the request.",
+            responses={
+            200: {
+                "content": {
+                    "application/json": {
+                        "example": response_example,
+                    }
+                },
+            },
+            422: {
+                "content": {
+                    "application/json": {
+                        "example": unprocessable_example
+                    }
+                }
+            }
+        },)
```

Comment:
Duplicate snippet/ code found in ocpJobs, can we define it in one place and import it here?

Reply:
I added some code to try to reduce some of this code duplication